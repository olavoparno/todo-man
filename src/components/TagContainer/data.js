export default [
  { id: 3, name: 'Food' },
  { id: 4, name: 'Tour' },
  { id: 5, name: 'Bills' },
  { id: 6, name: 'Housing' },
]
