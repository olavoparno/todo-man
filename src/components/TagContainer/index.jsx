import React from 'react'
import ReactTags from 'react-tag-autocomplete'
import PropTypes from 'prop-types'

import suggestionsData from './data'

import './styles.css'

class TagContainer extends React.PureComponent {

  constructor(props) {
    super(props)

    this.onChangeCb = props.onChangeCb

    this.state = {
      stateTags: props.todoTags || [],
      suggestions: suggestionsData,
    }
  }

  handleDelete = (i) => {
    const { stateTags } = this.state
    const tags = stateTags.slice(0)
    tags.splice(i, 1)

    this.onChangeCb(tags)

    return this.setState({ stateTags: tags })
  }

  handleAddition = (tag) => {
    const { stateTags } = this.state
    const tags = [].concat(stateTags, tag)

    this.onChangeCb(tags)

    return this.setState({ stateTags: tags })
  }

  render() {
    const { stateTags, suggestions } = this.state

    return (
      <ReactTags
        tags={stateTags}
        suggestions={suggestions}
        onDelete={index => this.handleDelete(index)}
        onAddition={tag => this.handleAddition(tag)}
      />
    )
  }

}

TagContainer.propTypes = {
  todoTags: PropTypes.array,
  onChangeCb: PropTypes.func,
}

export default TagContainer
