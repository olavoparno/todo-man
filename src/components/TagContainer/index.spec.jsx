import React from 'react'
import { shallow } from 'enzyme'

import TagContainer from './index'
import data from './data'

describe('TagContainer component', () => {
  let component
  beforeAll(() => {
    component = shallow(<TagContainer todoTags={data} />)
  })
  it('matches the snapshot', () => {
    expect(component).toMatchSnapshot()
  })
  it('should contain state todoTags', () => {
    expect(component.state().stateTags).toEqual(component.props().tags)
  })
})
