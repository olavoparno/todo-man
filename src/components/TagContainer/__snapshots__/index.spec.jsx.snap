// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagContainer component matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TagContainer
    todoTags={
      Array [
        Object {
          "id": 3,
          "name": "Food",
        },
        Object {
          "id": 4,
          "name": "Tour",
        },
        Object {
          "id": 5,
          "name": "Bills",
        },
        Object {
          "id": 6,
          "name": "Housing",
        },
        Object {
          "id": 7,
          "name": "Studies",
        },
        Object {
          "id": 8,
          "name": "Work",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "addOnBlur": false,
      "allowBackspace": true,
      "allowNew": false,
      "autoresize": true,
      "classNames": Object {
        "root": "react-tags",
        "rootFocused": "is-focused",
        "search": "react-tags__search",
        "searchInput": "react-tags__search-input",
        "searchWrapper": "react-tags__search-wrapper",
        "selected": "react-tags__selected",
        "selectedTag": "react-tags__selected-tag",
        "selectedTagName": "react-tags__selected-tag-name",
        "suggestionActive": "is-active",
        "suggestionDisabled": "is-disabled",
        "suggestions": "react-tags__suggestions",
      },
      "delimiters": Array [
        "Tab",
        "Enter",
      ],
      "id": "ReactTags",
      "inputAttributes": Object {},
      "maxSuggestionsLength": 6,
      "minQueryLength": 2,
      "onAddition": [Function],
      "onDelete": [Function],
      "placeholderText": "Add new tag",
      "removeButtonText": "Click to remove tag",
      "suggestionComponent": null,
      "suggestions": Array [
        Object {
          "id": 3,
          "name": "Food",
        },
        Object {
          "id": 4,
          "name": "Tour",
        },
        Object {
          "id": 5,
          "name": "Bills",
        },
        Object {
          "id": 6,
          "name": "Housing",
        },
        Object {
          "id": 7,
          "name": "Studies",
        },
        Object {
          "id": 8,
          "name": "Work",
        },
      ],
      "suggestionsFilter": null,
      "tagComponent": null,
      "tags": Array [
        Object {
          "id": 3,
          "name": "Food",
        },
        Object {
          "id": 4,
          "name": "Tour",
        },
        Object {
          "id": 5,
          "name": "Bills",
        },
        Object {
          "id": 6,
          "name": "Housing",
        },
        Object {
          "id": 7,
          "name": "Studies",
        },
        Object {
          "id": 8,
          "name": "Work",
        },
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "addOnBlur": false,
        "allowBackspace": true,
        "allowNew": false,
        "autoresize": true,
        "classNames": Object {
          "root": "react-tags",
          "rootFocused": "is-focused",
          "search": "react-tags__search",
          "searchInput": "react-tags__search-input",
          "searchWrapper": "react-tags__search-wrapper",
          "selected": "react-tags__selected",
          "selectedTag": "react-tags__selected-tag",
          "selectedTagName": "react-tags__selected-tag-name",
          "suggestionActive": "is-active",
          "suggestionDisabled": "is-disabled",
          "suggestions": "react-tags__suggestions",
        },
        "delimiters": Array [
          "Tab",
          "Enter",
        ],
        "id": "ReactTags",
        "inputAttributes": Object {},
        "maxSuggestionsLength": 6,
        "minQueryLength": 2,
        "onAddition": [Function],
        "onDelete": [Function],
        "placeholderText": "Add new tag",
        "removeButtonText": "Click to remove tag",
        "suggestionComponent": null,
        "suggestions": Array [
          Object {
            "id": 3,
            "name": "Food",
          },
          Object {
            "id": 4,
            "name": "Tour",
          },
          Object {
            "id": 5,
            "name": "Bills",
          },
          Object {
            "id": 6,
            "name": "Housing",
          },
          Object {
            "id": 7,
            "name": "Studies",
          },
          Object {
            "id": 8,
            "name": "Work",
          },
        ],
        "suggestionsFilter": null,
        "tagComponent": null,
        "tags": Array [
          Object {
            "id": 3,
            "name": "Food",
          },
          Object {
            "id": 4,
            "name": "Tour",
          },
          Object {
            "id": 5,
            "name": "Bills",
          },
          Object {
            "id": 6,
            "name": "Housing",
          },
          Object {
            "id": 7,
            "name": "Studies",
          },
          Object {
            "id": 8,
            "name": "Work",
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
